[DreamBooth] add text encoder LoRA support in the DreamBooth training script

sayakpaul opened the pull request:
Following up on https://github.com/huggingface/diffusers/pull/2918. Closes https://github.com/huggingface/diffusers/issues/3065

Hyperparameter tuning might change the game but with the default hyperparameters (taken from the LoRA training section of DreamBooth), it seems to be doing okay. 

Report: https://wandb.ai/sayakpaul/dreambooth-lora/reports/test-23-04-17-17-00-13---Vmlldzo0MDkwNjMy

Model repo (comes with a card if someone forgot): https://huggingface.co/sayakpaul/dreambooth

Sharing this cutie which was generated:
![image](https://user-images.githubusercontent.com/22957388/232474442-cfafbae5-ceb9-444e-9267-981f67c48c1f.png)

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/diffusers/pr_3130). All of your documentation changes will be reflected on that endpoint.

patrickvonplaten commented:
Great! Could we maybe also update the docs:
https://github.com/huggingface/diffusers/blob/main/docs/source/en/training/lora.mdx
and tests:
https://github.com/huggingface/diffusers/blob/main/examples/test_examples.py (think we've never added a lora test - my bad I added the dreambooth lora training script :sweat_smile: )

themrzmaster commented on examples/dreambooth/train_dreambooth_lora.py:
```diff
@@ -900,7 +948,11 @@ def main(args):
+                    params_to_clip = (
+                        itertools.chain(unet_lora_layers.parameters(), text_encoder_lora_layers.parameters())
+                        if args.train_text_encoder
+                        else text_encoder_lora_layers.parameters()
```
shouldn't it be 
`else text_encoder_lora_layers.parameters()`
?

sayakpaul commented:
@patrickvonplaten ready for a review now. I have addressed your comments. 

The tests were run using the command below:

```bash
python -m pytest -n 2 --max-worker-restart=0 --dist=loadfile -s -v -k "lora" examples/test_examples.py
```

innat-asj commented on docs/source/en/training/lora.mdx:
````diff
@@ -175,6 +177,11 @@ accelerate launch train_dreambooth_lora.py \
 ```
 
+It's also possible to additionally fine-tune the text encoder with LoRA. This, in most cases, leads
+to better results with a slight increase in the compute. To allow fine-tuning the text encoder with LoRA,
````
Would it be possible to add more experimental details of `unet_lora`  and `text_encoder_lora` (and combined) in separate blogs (like [dreambooth](https://huggingface.co/blog/dreambooth))?

sayakpaul replied:
We are leaving it to the community for now. But we will update if we find more :)

patrickvonplaten reviewed (approved):
Great job! Good to merge for me - we'll just need to resolve the merge conflicts

tcapelle commented:
Reverting to previous commits fix the issue

sayakpaul commented:
It's better if you create a new issue thread for this problems you are facing with a reproducible snippet.

tcapelle commented:
Yeah sorry, but I just pulled and integrated that commit and it broke, and realised that you just had merged that.
